File: src/Java/BehavioralPatterns/Command/FastFoodRestaurant/src/Commands/RemoveMenuItemCommand.java
package Commands;

import Models.MenuItem;

import java.util.Iterator;
import java.util.List;

public class RemoveMenuItemCommand extends OrderCommand {
    @Override
    public void execute(List<MenuItem> fastFoodOrderMenuItems, MenuItem menuItem) {
        Iterator<MenuItem> iterator = fastFoodOrderMenuItems.iterator();
        while (iterator.hasNext()) {
            MenuItem item = iterator.next();
            if (item.getName().equals(menuItem.getName())) {
                iterator.remove();
                break;
            }
        }
    }
}

File: src/PHP/BehavioralPatterns/Command/FastFoodRestaurant/Commands/RemoveMenuItemCommand.php
<?php

namespace BehavioralPatterns\Command\FastFoodRestaurant\Commands;

use BehavioralPatterns\Command\FastFoodRestaurant\Models\MenuItem;

class RemoveMenuItemCommand extends OrderCommand {
    public function execute(array &$fastFoodOrderMenuItems, MenuItem $menuItem): void
    {
        foreach ($fastFoodOrderMenuItems as $key => $fastFoodOrderMenuItem) {
            if ($fastFoodOrderMenuItem->name === $menuItem->name) {
                unset($fastFoodOrderMenuItems[$key]);
                break;
            }
        }

        $fastFoodOrderMenuItems = array_values($fastFoodOrderMenuItems);
    }
}